casper-deploy-notifier/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+use casper_types::AsymmetricType;
+use eventsource_stream::{Event, EventStreamError, Eventsource};
+use futures::stream::{BoxStream, TryStreamExt};
+use tokio::sync::mpsc;
+
+use crate::error::SseError;
+use crate::sse_types::SseData;
+use crate::types::Notification;
+
+mod error;
+mod sse_types;
+mod types;
+
+const DEFAULT_SSE_SERVER: &str = "https://events.mainnet.casperlabs.io";
+const DEFAULT_EVENT_CHANNEL: &str = "/events/main";
+
+type BoxedEventStream = BoxStream<'static, Result<Event, EventStreamError<reqwest::Error>>>;
+
+pub struct DeployNotifier {
```

Comment:
I don't see a point why this even needs to be a type. A single function:
```Rust
async pub fn subscribe_to_node(url: &reqwest::URL, sender: mpsc::Sender<DeployProcessed>) {
  let stream = connect(url);
  ...
  sender.send(deploy_processed)
  // or if you want to continue to use Notification: sender.send(deploy_processed.into())
  ...
}
```

Reply:
I could convert `DeployNotifier` to a single function now, but I prefer to have code prepared for future improvements, e.g. filtering, queuing, timeouts.

Reply:
I like it as a method

Reply:
What do you mean with method? an associated function?
Can you explain why you prefer a type here?

Reply:
Method is associated function.

Too much abstraction can be overkill sometimes, but it does not hurt us, especially with such simple struct that is promising to evolve. Again, think about Casper community, not just Kairos.

Reply:
Both could work fine for now, I don' really care which, but I expect this to grow into the builder pattern.